openmls/src/group/core_group/apply_proposals.rs

```diff
@@ -168,15 +164,16 @@ impl CoreGroup {
 
         let presharedkeys = PreSharedKeys { psks: psks.into() };
 
-        // This flag determines if the commit requires a path. A path is
-        // required if the commit is empty, i.e. if it doesn't contain any
-        // proposals or if it is a "full" commit. A commit is full if it refers
-        // to proposal types other than Add, PreSharedKey and/or ReInit
-        // proposals.
-        let path_required = has_updates
-            || has_removes
-            // The fact that this is some implies that there's an external init
-            // proposal.
+        let proposals_require_path = proposal_queue
+            .queued_proposals()
+            .any(|p| p.proposal().is_path_required());
```

Comment:
The drawback with this approach is that we have to iterate over all proposals again. As opposed to just setting a flag when we're iterating over them anyway. I don't think it'll have that much of an impact, though, so it's probably fine.

Reply:
Yes I think it's fine, it does not add that much overhead. The biggest problem here would be to miss some proposal type (for example if it's not implemented yet like GCE proposals).

Anyway in this method `proposal_queue.filtered_by_type` already iterates over the proposal queue. I think the `CoreGroup::apply_proposals` method should be rewritten like this to do what you suggest.

```rust
let mut is_path_required = false;
proposal_queue
    .queued_proposals()
    .into_iter()
    .for_each(|p| {
        is_path_required |= p.proposal().is_path_required();
        match p.proposal() {
            Proposal::Add(add_proposal) => {
                todo!()
            }
            _ => todo!(),
        }
    });
```